Enhanced the documentation for the convert_index_to_remote ISM action with comprehensive details and examples: #11491 (#11495)
opensearch-project/documentation-website: _im-plugin/ism/policies.md

### convert_index_to_remote

Converts an index to a searchable snapshot by restoring it from a remote snapshot repository. This action is useful for reducing storage costs by moving infrequently accessed data to remote storage while keeping it searchable. The action performs a restore operation from the specified snapshot and automatically deletes the original index once the restore is successfully accepted, ensuring only the remote index remains.

The `convert_index_to_remote` operation has the following parameters.

Parameter | Description | Type | Required | Default
:--- | :--- |:--- |:--- |

> `repository` | The repository name registered through the native snapshot API operations. Must be a remote repository type (for example, S3, Azure, or GCS).  | `string` | Yes | N/A

Reviewer: What are the actual valid values for this parameter? Is it `s3`, `azure`, and `gcs`? Please list them in code font.

`snapshot` | The snapshot name created through the snapshot action.  | `string` | Yes | N/A
`include_aliases` | Whether to include index aliases during the restore operation. When set to `true`, all aliases associated with the original index are restored along with the remote index. Set to `true` if your application references the index through aliases. | `boolean` | No | `false`

> `ignore_index_settings` | A comma-separated list of index settings to ignore during the restore operation. For example, `"index.refresh_interval,index.number_of_replicas"`. This is useful when you want to apply different settings to the restored remote index than what was configured in the original index. | `string` | No | Empty string

Reviewer: ```suggestion
`ignore_index_settings` | A comma-separated list of index settings to ignore during the restore operation. For example, `index.refresh_interval,index.number_of_replicas`. This is useful when you want to apply different settings to the restored remote index than the ones configured in the original index. | String | No | Empty string
```

`number_of_replicas` | The number of replicas to configure for the restored remote index. This allows you to control replica allocation during the conversion process without requiring a separate update operation. Setting this during conversion helps prevent cluster yellow state or unnecessary load from replica allocation. | `integer` | No | `0`

#### Prerequisites

Before using the `convert_index_to_remote` action, ensure the following:

- A remote repository (S3, Azure, or GCS) is registered and accessible.
- A snapshot of the index exists in the specified repository, typically created using the `snapshot` action.
- The repository name matches the one used in the snapshot action.

#### Usage notes

- **Automatic deletion**: The original index is automatically deleted after the remote snapshot restore is successfully accepted. This ensures that only the searchable snapshot version remains, completing the conversion process.

> - **Repository matching**: The repository name used in the `convert_index_to_remote` operation must match the repository name specified during the snapshot action.

Reviewer: ```suggestion
- The repository name used in the `convert_index_to_remote` operation must match the repository name specified during the snapshot action.
```
